src/render/tutorials/chrome-plugin/2_2.html.md

```diff
@@ -0,0 +1,495 @@
+  factory = http
+  config.endpointOptions {
+    path = /offers
+    domain = webapi
+    port = 8080
+  }
+}
+```
+
+When we refresh the page we still won't see any offers displayed on the page. We'll get to this in a minute. The graph should look like this:
+
+![Fallback](/img/blog/chrome-extension/with-fallback.png)
+
+As we can see, even though the circuit breaker ended with fallback, the composite as a whole ended with success because of the `fetch-offers-fallback` node.
+
+The only problem left is that we can't see any offers displayed, even though they were retrieved (you can check they were by viewing details of the fallback node). Let's take a look at the HTML snippet:
+
+```html
+<knotx:snippet data-knotx-task="payments-task">
+  <div>
+    <h2>Payment providers</h2>
+    <ul>
+      {% for provider in data['fetch-payment-providers']._result %}
+      <li>
+        <p>Payment provider: {{provider.name}}</p>
+        <p>Payment URL: {{provider.paymentUrl}}</p>
+      </li>
+      {% endfor %}
+    </ul>
+
+    <h2>Offers</h2>
+    <ul>
+      {% for offer in data['fetch-offers']._result %}
+      <li>
+        <p>Offer: {{offer.method}}</p>
+        <p>Price: {{offer.price}}</p>
+      </li>
+      {% endfor %}
+    </ul>
+  </div>
+</knotx:snippet>
+```
+
+Pebble retrieves offers data from a key called `fetch-offers`. By default, every action saves its data under its name. The problem is that in case of a timeout the action that retrieves data is called `fetch-offers-fallback` and therefore that's where the data is stored. One way to resolve this would be to use an action that rewrites data from one key to another. For the sake of this tutorial we'll just create additional markup for fallback messages. The Offers section of the snippet will look like this:
+
+```html
+<h2>Offers</h2>
+<ul>
+  {% for offer in data['fetch-offers']._result %}
+  <li>
+    <p>Offer: {{offer.method}}</p>
+    <p>Price: {{offer.price}}</p>
+  </li>
+  {% endfor %}
+  {% for offer in data['fetch-offers-fallback']._result %}
+  <li>
+    <p>Fallback offer: {{offer.method}}</p>
+    <p>Fallback price: {{offer.price}}</p>
+  </li>
+  {% endfor %}
+</ul>
+```
+
+If we render the page now, we'll be able to see the fallback offers:
+
+![Fallback Offers](/img/blog/chrome-extension/fallback-offers.png)
+
+## Summary
+As you can see Knot.x Fragments allows us to easily define data flow when integrating with 3rd party services. In this example we rendered a page, however this mechanism can be used to build restful APIs as well. The Knot.x extension can be used during many processes of building an application. It can be used by developers for debugging purposes, new team members to familiarize themselves with the application's architecture, quality assurance/testers to detect flaws and identify root causes of bugs (is it a frontend, backend or integration problem?) or architects and business people to visualize the data flow and buisiness cases.

```

Comment:
```suggestion
As you can see Knot.x Fragments allows us to easily define data flow when integrating with 3rd party services. In this example, we rendered a page, but note this mechanism can be used to build restful APIs as well. The Knot.x extension can be used during many processes of building an application. It can be used by developers for debugging purposes, new team members to familiarize themselves with the application's architecture, quality assurance/testers to detect flaws and identify root causes of bugs (is it a frontend, backend or integration problem?) or architects and business people to visualize the data flow and business cases.
```